osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -485,16 +485,32 @@ private void addControlPoint(double time, ControlPoint point, bool timingChange)
         private void flushPendingPoints()
         {
             // Changes from non-timing-points are added to the end of the list (see addControlPoint()) and should override any changes from timing-points (added to the start of the list).
+            // As non-timing-points should not set the OmitFirstBarline setting, we will honor the setting of the last effect point, the effect point generated by the timing-point (if any).
```

Comment:
This doesn't hold true in lazer, where you can actually set the setting on a non-timing point. I honestly don't know how valid that is, but it's a thing..

Reply:
The todo is specifically related to the stable beatmap format points that get fed into this function in the first place, from which we should honor the ones first in the list i.e. last in the loop more because they're effect points originating from timing points. Fixing that non timing points shouldnt be able to set this anyway might be an issue for another day.

Reply:
For sure. But as you had it this also would break lazer editor behaviour. Which can be seen by the first case in your tests (which asserts `false` but should probably assert `true`).